### PR TITLE
Converting ph commands

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

ph commands has been converted.

i added `phl` for listing all algorithms

### CODE REVIEW

File: librz/core/cmd_descs/cmd_print.yaml [hunk 1]
```diff
@@ -48,6 +48,29 @@ commands:
...
+    cname: cmd_print_default
+    subcommands:
+      - name: ph
+        summary: Prints a hash/message digest or entropy
+        cname: cmd_print_msg_digest
+        args:
+          - name: algo
+            type: RZ_CMD_ARG_TYPE_STRING
+          - name: length
+            type: RZ_CMD_ARG_TYPE_NUM
```
Comment (ret2libc) [inline librz/core/cmd_descs/cmd_print.yaml]:
```suggestion
            type: RZ_CMD_ARG_TYPE_RZNUM
```

File: librz/core/cmd_print.c [hunk 2]
```diff
@@ -3207,59 +3207,46 @@ static void disasm_strings(RzCore *core, const char *input, RzAnalysisFunction *
...
-			} else {
-				handle_msg_digest(plugin->name, core->block, len);
-			}
-			handled_cmd = true;
-			break;
+
+	ut32 old_blocksize = core->blocksize;
+	ut32 new_blocksize = old_blocksize;
+
+	if (RZ_STR_ISNOTEMPTY(argv[1])) {
```
Comment (ret2libc) [inline librz/core/cmd_print.c]: argv[1] cannot be null, so you can avoid this `if`.

File: librz/core/cmd_print.c [hunk 3]
```diff
@@ -3207,59 +3207,46 @@ static void disasm_strings(RzCore *core, const char *input, RzAnalysisFunction *
...
+		plugin = rz_msg_digest_plugin_by_name(argv[1]);
+	}
+
+	if (!plugin) {
+		RZ_LOG_ERROR("algorithm '%s' does not exists\n", argv[1]);
+		return RZ_CMD_STATUS_INVALID;
```
Comment (ret2libc) [inline librz/core/cmd_print.c]:
```suggestion
		return RZ_CMD_STATUS_ERROR;
```
I think this is better. _INVALID is used mainly for shell/parser/grammar errors.

File: librz/core/cmd_print.c [hunk 4]
```diff
@@ -3207,59 +3207,46 @@ static void disasm_strings(RzCore *core, const char *input, RzAnalysisFunction *
...
+	}
+
+	if (argc == 3) {
+		if (!rz_num_is_valid_input(core->num, argv[2])) {
+			return RZ_CMD_STATUS_INVALID;
```
Comment (ret2libc) [inline librz/core/cmd_print.c]: ERROR as well.

File: librz/core/cmd_print.c [hunk 5]
```diff
@@ -3207,59 +3207,46 @@ static void disasm_strings(RzCore *core, const char *input, RzAnalysisFunction *
...
 		}
+		new_blocksize = rz_num_math(core->num, argv[2]);
+
+		if (!rz_core_block_size(core, new_blocksize)) {
+			rz_core_block_size(core, old_blocksize);
```
Comment (ret2libc) [inline librz/core/cmd_print.c]: this is likely not necessary if the previous call has failed.

File: librz/core/cmd_descs/cmd_print.yaml [hunk 6]
```diff
@@ -48,6 +48,29 @@ commands:
...
+            optional: true
```
Comment (ret2libc) [inline librz/core/cmd_descs/cmd_print.yaml]: In theory we could eliminate these kind of things and just rely on `@!`, which works for every command. If we use that approach we can assume all commands work on core->offset and with core->blocksize, but you can always use `@` and `@!` to change those. Just an idea anyway...

File: librz/core/cmd_print.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd_print.c:3216]: i think i made a mistake. that was supposed to be the number.

File: librz/core/cmd_descs/cmd_print.yaml (hunk 6 above)
Comment (wargio) [reply librz/core/cmd_descs/cmd_print.yaml]: i agree. i thought about that too.. but usability may sink..

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**